components/dm/command/scale_in.go

```diff
@@ -100,23 +99,17 @@ func ScaleInDMCluster(
 	}
 
 	// Cannot delete all DM DMMaster servers
-	if len(deletedDiff[dm.ComponentDMMaster]) == len(spec.Masters) {
+	if len(deletedDiff[dm.ComponentDMMaster]) == len(topo.Masters) {
 		return errors.New("cannot delete all dm-master servers")
 	}
 
 	if options.Force {
-		for _, component := range spec.ComponentsByStartOrder() {
+		for _, component := range topo.ComponentsByStartOrder() {
 			for _, instance := range component.Instances() {
 				if !deletedNodes.Exist(instance.ID()) {
 					continue
 				}
-				// just try stop and destroy
-				if err := operator.StopComponent(getter, []dm.Instance{instance}, options.OptTimeout); err != nil {
-					log.Warnf("failed to stop %s: %v", component.Name(), err)
-				}
-				if err := operator.DestroyComponent(getter, []dm.Instance{instance}, spec, options); err != nil {
-					log.Warnf("failed to destroy %s: %v", component.Name(), err)
-				}
+				_ = operator.StopAndDestroyInstance(getter, component.Name(), topo, instance, options, false)
```

Comment:
We shouldn't ignore the error, please use `log.Warnf` if this error is not critical.